scripts/cli.sh

```diff
@@ -0,0 +1,107 @@
+#! /usr/bin/env bash
+
+set -e
+
+# A function to generate formatted .md files
+write_to_file() {
+    local cmd="$1"
+    local file="$2"
+    local program="$3"
+
+    # Remove first line of cmd to get rid of commit specific numbers.
+    cmd=${cmd#*$'\n'}
+
+    # We need to add the header and the backticks to create the code block.
+    printf "# %s\n\n\`\`\`\n%s\n\`\`\`" "$program" "$cmd" > "$file"
+}
+
+# Check if a lighthouse binary exists in the current branch.
+# -f means check if the file exists, to see all options, type "bash test" in a terminal
+release=./target/release/lighthouse
+debug=./target/debug/lighthouse
+
+if [[ -f "$release" ]]; then
+    CMD="$release"
+elif [[ -f "$debug" ]]; then
+    CMD="$debug"
+else
+    # No binary exists, build it.
+    cargo build --locked
+    CMD="$debug"
+fi
+
+# Store all help strings in variables.
+general_cli=$($CMD --help)
+bn_cli=$($CMD bn --help)
+vc_cli=$($CMD vc --help)
+vm_cli=$($CMD vm --help)
+vm_cli_create=$($CMD vm create --help)
+vm_cli_import=$($CMD vm import --help)
+vm_cli_move=$($CMD vm move --help)
+
+general=./help_general.md
+bn=./help_bn.md
+vc=./help_vc.md
+am=./help_am.md
+vm=./help_vm.md
+vm_create=./help_vm_create.md
+vm_import=./help_vm_import.md
+vm_move=./help_vm_move.md
+
+# create .md files
+write_to_file "$general_cli" "$general" "Lighthouse General Commands"
+write_to_file "$bn_cli" "$bn" "Beacon Node"
+write_to_file "$vc_cli" "$vc" "Validator Client"
+write_to_file "$vm_cli" "$vm" "Validator Manager"
+write_to_file "$vm_cli_create" "$vm_create" "Validator Manager Create"
+write_to_file "$vm_cli_import" "$vm_import" "Validator Manager Import"
+write_to_file "$vm_cli_move" "$vm_move" "Validator Manager Move"
+
+#input 1 = $1 = files; input 2 = $2 = new files
+files=(./book/src/help_general.md ./book/src/help_bn.md ./book/src/help_vc.md ./book/src/help_vm.md ./book/src/help_vm_create.md ./book/src/help_vm_import.md ./book/src/help_vm_move.md)
+new_files=($general $bn $vc $vm $vm_create $vm_import $vm_move)
+
+# function to check
+check() {
+    local file="$1"
+    local new_file="$2"
+    
+if [[ -f $file ]]; then # check for existence of file 
+    diff=$(diff $file $new_file)
```

Comment:
Ahh, if `diff` contains a change, it will actually return an error code of 1.
See `diff --help`:
```
...
Exit status is 0 if inputs are the same, 1 if different, 2 if trouble.
...
```

Since we now have `set -e`, if any part of our script raises a non-zero exit code, the script will exit early. Since we still want `set -e` for the majority of cases, we just want to suppress this particular error code.

We can do that by adding `|| :` to the end of the diff command. Basically this says, "in the event of an error, do nothing", and thus will not raise the error.
So it should look like:
```suggestion
    diff=$(diff $file $new_file || :)
```

--- 
(I believe CI didn't catch this since the files do not exist yet and so the first CI run will avoid this particular code path. I think every future CI run would have run into this bug.)

Reply:
This is a really great point, thank you!